test/integration/short/client-pool-tests.js

```diff
@@ -369,6 +369,45 @@ describe('Client', function () {
         helper.ccmHelper.remove
       ], done);
     });
+    function getReceiveNotificationTest(nodeNumber) {
+      return (function receiveNotificationTest(done) {
+        // Should receive notification when a node gracefully closes connections
+        var client = newInstance({
+          pooling: {
+            warmup: true,
+            heartBeatInterval: 0,
+            // Use just 1 connection per host for all protocol versions in this test
+            coreConnectionsPerHost: clientOptions.coreConnectionsPerHostV3
+          }
+        });
+        utils.series([
+          helper.ccmHelper.removeIfAny,
+          helper.ccmHelper.start(2),
+          client.connect.bind(client),
+          function checkInitialState(next) {
+            var hosts = client.hosts.values();
+            assert.ok(hosts[0].isUp());
+            assert.ok(hosts[1].isUp());
+            next();
+          },
+          function stopNode(next) {
+            helper.ccmHelper.stopNode(nodeNumber, next);
+          },
+          helper.delay(300),
```

Comment:
Is the 300ms delay needed?  From what I can tell the driver should react quickly enough (usually before the process exists as result of stopNode).